src/main/java/org/opensearch/neuralsearch/processor/normalization/ScoreNormalizationUtil.java

```diff
@@ -82,4 +82,78 @@ public static void setNormalizedScore(
         }
         scores.set(subQueryIndex, normalizedScore);
     }
+
+    /**
+     * Validate parameters for this technique. Following is example of structured parameters that we will validate
+     * {
+     *     "technique": "arithmetic_mean",  // top-level parameter 1
+     *     "details": {                     // top-level parameter 2
+     *         "weights": [1, 2, 3],        // nested parameter 1
+     *         "color": "green".            // nested parameter 2
+     *     }
+     * }
+     * for this example client should pass:
+     *     top-level parameters: ["technique", "details"]
+     *     nested parameters: ["details" -> ["weights", "color"]]
+     * @param actualParameters map of actual parameters in form of name-value
+     * @param supportedParametersTopLevel collection of top-level parameters that we should validate against
+     * @param supportedParametersNested map of nested parameters that we should validate against, key is one of top level parameters and value is set of allowed nested params
+     */
+    public void validateParameters(
+        final Map<String, Object> actualParameters,
+        final Set<String> supportedParametersTopLevel,
+        final Map<String, Set<String>> supportedParametersNested
+    ) {
+        if (Objects.isNull(actualParameters) || actualParameters.isEmpty()) {
+            return;
+        }
+        boolean hasUnknownParameters = false;
+        for (Map.Entry<String, Object> entry : actualParameters.entrySet()) {
+            String paramName = entry.getKey();
+            Object paramValue = entry.getValue();
+
+            if (!supportedParametersTopLevel.contains(paramName)) {
```

Comment:
```suggestion
            if (supportedParametersTopLevel.contains(paramName)==false) {
```

Reply:
same as above, I do like the minimalistic syntax